### DISCUSSION TITLE
Store *.pyc files outside the app bundle

### DISCUSSION THREAD

**Discussion by panathea** [opensteno/plover#1236] [Ideas]
If we ever want to code sign the app bundle on Mac, we can't have [`*.pyc` files generate](https://developer.apple.com/forums/thread/75060) in the embedded Python.framework.

Code signing takes into account all the contents of the app bundle. Currently on first run, python compiles all the byte-code and places it alongside the framework in the app bundle, which would invalidate the signature.

We should investigate whether [PYTHONCACHEPREFIX](https://stackoverflow.com/a/471985/3254906) (python3.8+) could be used to store the pyc files in ~/Library/Application Support/Open Steno Project/Plover` instead.

**Comment by benoit-pierre** [comment]
Don't we already remove (most) Python source files? (via `plover_build_utils.source_less`)

Is the issue with the Python modules we keep as source?

**Reply by benoit-pierre** [reply]
(And make sure we compile those, even if we keep them).

**Reply by panathea** [reply]
Hmm, you're right. For fun, I took a copy of Plover freshly downloaded, backed it up, and ran the original and closed Plover.

Looks like the folders are the same size on disk and `diff -r Plover.app Plover2.app` returned nothing. So this may be a nonissue!

**Reply by benoit-pierre** [reply]
We don't compile those source files we keep:
```bash
> fd '_in_process.*' plover-4.0.0.dev8+369.gd324d4e-macosx_10_13_x86_64
plover-4.0.0.dev8+369.gd324d4e-macosx_10_13_x86_64/Plover/Plover.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py
```
If we don't fix that, I except some other fun issues when installing plugins.

**Reply by benoit-pierre** [reply]
We could also just keep the sources, and make sure we pre-compile everything to bytecode. I don't remember how much was saved by making distributions source-less. But now that we don't include Qt's WebEngine, we have some room to spare.
